### ISSUE TITLE
[Bug Report] App ist crashing

### ISSUE THREAD

**Issue by Lauritz-Tieste** [SapuSeven/BetterUntis#362]
### Steps to Reproduce
Click on any element, swipe, do something

### Expected Result

The app shouldn't crash 

### Actual Result

The app is crashing

### Android Version

Android 13

### BetterUntis Version

4.0.1 beta3

### Installation Source

Google Play

**Comment by Lauritz-Tieste** [comment]
https://user-images.githubusercontent.com/84938977/220080311-eedb8d1e-aad6-4ca6-b388-39238629cf74.mp4

**Comment by SapuSeven** [comment]
Hard to say what's going on here with just the video. Do you have access to a crash log?

**Comment by Lauritz-Tieste** [comment]
how can I get the crash report?

**Comment by SapuSeven** [comment]
Do you get the "App crashed" dialog? If you installed the app from google play, you should see the option "send feedback". There you can scroll down to "stack trace".

Otherwise, you would need to download a logcat reader app and search for "AndroidRuntime".

**Comment by Lauritz-Tieste** [comment]
I found the cause of the error.  I have activated two factor authentication. This was the error.

**Comment by lm41** [comment]
In #239 someone stated that you can use the app with 2FA enabled, but that you have to use your untis mobile key instead of the password.

**Comment by Lauritz-Tieste** [comment]
the two factor authentication doesn't seem to have been the problem.  I gave a bug report with the "Give feedback" dialog.

**Comment by SapuSeven** [comment]
> the two factor authentication doesn't seem to have been the problem. I gave a bug report with the "Give feedback" dialog.

Unfortunately I can't find the report.
Could you please attach a Screenshot of the Stack Trace in the report dialog to this issue directly?

**Comment by Lauritz-Tieste** [comment]
This message was displayed in the logcat more than 300 times after the App crashed

Negative delta from freq time for uid: 99501, delta: -317

**Comment by Lauritz-Tieste** [comment]
This message was displayed more than 300 times after the App crashed

Negative delta from freq time for uid: 99501, delta: -317

**Comment by Lauritz-Tieste** [comment]
I have sent another report

**Comment by SapuSeven** [comment]
```
Exception android.database.sqlite.SQLiteDatabaseLockedException: database is locked (code 5 SQLITE_BUSY[5])
  at android.database.sqlite.SQLiteConnection.nativeExecute (SQLiteConnection.java)
  at android.database.sqlite.SQLiteConnection.execute (SQLiteConnection.java:999)
  at android.database.sqlite.SQLiteSession.beginTransactionUnchecked (SQLiteSession.java:321)
  at android.database.sqlite.SQLiteSession.beginTransaction (SQLiteSession.java:300)
  at android.database.sqlite.SQLiteDatabase.beginTransaction (SQLiteDatabase.java:954)
  at android.database.sqlite.SQLiteDatabase.beginTransactionNonExclusive (SQLiteDatabase.java:888)
  at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.beginTransactionNonExclusive (FrameworkSQLiteDatabase.kt:59)
  at androidx.room.RoomDatabase.internalBeginTransaction (RoomDatabase.kt:524)
  at androidx.room.RoomDatabase.beginTransaction (RoomDatabase.kt:508)
  at com.sapuseven.untis.data.databases.entities.UserDao_Impl.getByIdWithData (UserDao_Impl.java:1792)
  at com.sapuseven.untis.activities.MainActivityKt$WeekViewCompose$2.invoke (MainActivity.kt:186)
  at com.sapuseven.untis.activities.MainActivityKt$WeekViewCompose$2.invoke (MainActivity.kt:170)
  at androidx.compose.ui.viewinterop.ViewFactoryHolder$updateBlock$1.invoke (AndroidView.android.kt:186)
  at androidx.compose.runtime.snapshots.Snapshot$Companion.observe (Snapshot.kt:2200)
  at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke (SnapshotStateObserver.kt:234)
  at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke (SnapshotStateObserver.kt:230)
  at androidx.compose.runtime.SnapshotStateKt__DerivedStateKt.observeDerivedStateRecalculations (SnapshotStateKt__DerivedStateKt.java:341)
  at androidx.compose.runtime.SnapshotStateKt.observeDerivedStateRecalculations (SnapshotStateKt.java:1)
  at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads (SnapshotStateObserver.kt:230)
  at androidx.compose.ui.viewinterop.AndroidViewHolder$runUpdate$1.invoke (AndroidViewHolder.android.kt:170)
  at androidx.compose.ui.viewinterop.AndroidViewHolder$runUpdate$1.invoke (AndroidViewHolder.android.kt:168)
  at androidx.compose.ui.viewinterop.AndroidViewHolder.setUpdate (AndroidViewHolder.android.kt:108)
  at androidx.compose.ui.viewinterop.ViewFactoryHolder.setUpdateBlock (AndroidView.android.kt:186)
  at androidx.compose.ui.viewinterop.AndroidView_androidKt$AndroidView$2$5.invoke (AndroidView.android.kt:130)
  at androidx.compose.runtime.ComposerImpl$apply$operation$1.invoke (Composer.kt:1696)
  at androidx.compose.runtime.ComposerImpl$apply$operation$1.invoke (Composer.kt:1694)
  at androidx.compose.runtime.CompositionImpl.applyChangesInLocked (Composition.kt:810)
  at androidx.compose.runtime.CompositionImpl.applyChanges (Composition.kt:842)
  at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke (Recomposer.kt:592)
  at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke (Recomposer.kt:510)
  at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame (AndroidUiFrameClock.android.kt:34)
  at androidx.compose.ui.platform.AndroidUiDispatcher.performFrameDispatch (AndroidUiDispatcher.java:109)
  at androidx.compose.ui.platform.AndroidUiDispatcher.access$performFrameDispatch (AndroidUiDispatcher.java:41)
  at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame (AndroidUiDispatcher.android.kt:69)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:1299)
  at android.view.Choreographer$CallbackRecord.run (Choreographer.java:1309)
  at android.view.Choreographer.doCallbacks (Choreographer.java:923)
  at android.view.Choreographer.doFrame (Choreographer.java:847)
  at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:1283)
  at android.os.Handler.handleCallback (Handler.java:942)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8741)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:571)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1067)
```

**Comment by lm41** [comment]
https://stackoverflow.com/a/63849407
I don't know if this works.

**Comment by Lauritz-Tieste** [comment]
A reinstall fixed the error

**Comment by lm41** [comment]
Fixed with 86abf6b2d9937ad3b12beb39b604f12accccadb3
